lib/utils/createDomain.js

```diff
@@ -2,30 +2,24 @@
 
 const url = require('url');
 
-function createDomain(options, server) {
+function createDomain(options) {
   const protocol = options.https ? 'https' : 'http';
-  // use location hostname and port by default in createSocketURL
-  // ipv6 detection is not required as 0.0.0.0 is just used as a placeholder
-  let hostname;
-
-  if (server) {
-    hostname = server.address().address;
-  } else {
-    hostname = '0.0.0.0';
-  }
-
-  const port = server ? server.address().port : 0;
 
   // use explicitly defined public url
   // (prefix with protocol if not explicitly given)
-  if (options.public) {
-    return /^[a-zA-Z]+:\/\//.test(options.public)
-      ? `${options.public}`
-      : `${protocol}://${options.public}`;
+  if (options.client.socketUrl) {
```

Comment:
```suggestion
  if (options.client && options.client.socketUrl) {
```

Reply:
We set `options.client` to `{}` in `normalizeOptions`

Reply:
All good then 👍🏻